components/LocationInput.tsx

```diff
@@ -66,27 +63,40 @@ const LocationInput: FC<LocationInputProps> = ({
 
       let popup: mapboxgl.Popup | null = null;
 
+      const renderFunction = (item: MapboxGeocoder.Result) => {
+        var placeName = item.place_name.split(',');
+        return canEdit
+          ? '<div class="mapboxgl-ctrl-geocoder--suggestion"><div class="mapboxgl-ctrl-geocoder--suggestion-title">' +
+              placeName[0] +
+              '</div><div class="mapboxgl-ctrl-geocoder--suggestion-address">' +
+              placeName.splice(1, placeName.length).join(',') +
+              '</div></div>'
+          : '';
+      };
+
       const geocoder = new MapboxGeocoder({
         accessToken: mapboxgl.accessToken,
         mapboxgl: mapboxgl,
+        countries: 'GB',
         placeholder: 'Search a location',
         marker: false,
+        render: renderFunction,
         reverseGeocode: true,
+        limit: 3,
       });
 
       if (!canEdit) {
         // Prevent user from changing the geocoder input when the map isn't editable
         geocoder.setBbox([0, 0, 0, 0]);
       }
 
-      geocoderContainer.current?.appendChild(geocoder.onAdd(map.current));
-
       const marker = new mapboxgl.Marker({
         draggable: false,
         color: '#FFD053',
       }).setLngLat([originalLat, originalLng]);
 
       map.current.on('load', () => {
+        geocoderContainer.current?.appendChild(geocoder.onAdd(map.current!));
```

Comment:
What does moving this to after load do?

Reply:
Nothing—moved it back outside, since it looks better to have the geocoder and the map render at the same time. This was just a vestige from one of my attempted bug fix methods.